components/purchase_modal/purchase_modal.tsx

```diff
@@ -44,17 +47,36 @@ type Props = {
         closeModal: () => void;
         getCloudProducts: () => void;
         completeStripeAddPaymentMethod: (stripe: Stripe, billingDetails: BillingDetails, isDevMode: boolean) => Promise<boolean | null>;
+        updateCloudSubscription: (productId: string) => Promise<boolean | null>;
         getClientConfig: () => void;
         getCloudSubscription: () => void;
     };
 }
 
 type State = {
     paymentInfoIsValid: boolean;
-    productPrice: number;
     billingDetails: BillingDetails | null;
     cardInputComplete: boolean;
     processing: boolean;
+    selectedProduct: Product | null | undefined;
+}
+
+function findProductInDictionary(products: Dictionary<Product>, productId?: string): Product | null {
+    const keys = Object.keys(products);
+    let selectedProduct = null;
+    if (keys.length > 1) {
+        // here find the product by the provided id, otherwise return the one with Professional in the name
+        keys.forEach((key) => {
+            if (productId && products[key].id === productId) {
+                selectedProduct = products[key];
+            } else if (!productId && products[key].name.includes('Professional')) {
+                selectedProduct = products[key];
+            }
+        });
+    } else if (keys.length === 1) {
+        selectedProduct = products[keys[0]];
+    }
```

Comment:
Related to the comment above, we should just do this if no selected product is found above.